project/util/mailing_address.py

```diff
@@ -8,6 +9,15 @@
 US_STATE_CHOICES = Choices.from_file('us-state-choices.json')
 
 
+class ZipCodeValidator(RegexValidator):
+    def __init__(self):
+        # https://www.oreilly.com/library/view/regular-expressions-cookbook/9781449327453/ch04s14.html
+        super().__init__(
+            regex=r"^[0-9]{5}(?:-[0-9]{4})?$",
+            message="Enter a valid U.S. zip code."
+        )
```

Comment:
I split this out into its own class that takes no arguments to ensure that changes to the regex or error message don't trigger trivial migrations.